### DISCUSSION TITLE
Trying to build strongswan6.0 beta2 release with liboqs library for oqs plugin

### DISCUSSION THREAD

**Discussion by abc1233user** [strongswan/strongswan#1531] [Q&A] [answered]
I am unable to correctly specify how to include installed liboqs library with ./configure of strongswan6.0 beta2 on ubuntu 22.04

I installed liboqs and ran the example files [here](https://github.com/open-quantum-safe/liboqs/wiki/Minimal-example-of-a-post-quantum-KEM) to verify correct installation of liboqs.

Now after extracting files from _strongswan-6.0beta2.tar.bz2_ and running: 
`
 LIBS=-loqs CFLAGS=-I/home/umair/liboqs/build/include LDFLAGS=-L/home/umair/liboqs/build/lib ./configure --prefix=/usr/local --enable-oqs` 
and  `make` I get the following errors:
 
> `oqs_kem.c: In function ‘oqs_kem_create’:
> oqs_kem.c:245:35: error: ‘OQS_KEM_alg_ntru_hps2048509’ undeclared (first use in this function)
>   245 |                         kem_alg = OQS_KEM_alg_ntru_hps2048509;
>       |                                   ^~~~~~~~~~~~~~~~~~~~~~~~~~~
> oqs_kem.c:245:35: note: each undeclared identifier is reported only once for each function it appears in
> oqs_kem.c:248:35: error: ‘OQS_KEM_alg_ntru_hps2048677’ undeclared (first use in this function)
>   248 |                         kem_alg = OQS_KEM_alg_ntru_hps2048677;
>       |                                   ^~~~~~~~~~~~~~~~~~~~~~~~~~~
> oqs_kem.c:251:35: error: ‘OQS_KEM_alg_ntru_hps4096821’ undeclared (first use in this function)
>   251 |                         kem_alg = OQS_KEM_alg_ntru_hps4096821;
>       |                                   ^~~~~~~~~~~~~~~~~~~~~~~~~~~
> oqs_kem.c:254:35: error: ‘OQS_KEM_alg_ntru_hrss701’ undeclared (first use in this function)
>   254 |                         kem_alg = OQS_KEM_alg_ntru_hrss701;
>       |                                   ^~~~~~~~~~~~~~~~~~~~~~~~
> oqs_kem.c:257:35: error: ‘OQS_KEM_alg_saber_lightsaber’ undeclared (first use in this function)
>   257 |                         kem_alg = OQS_KEM_alg_saber_lightsaber;
>       |                                   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
> oqs_kem.c:260:35: error: ‘OQS_KEM_alg_saber_saber’ undeclared (first use in this function); did you mean ‘OQS_KEM_alg_kyber_768’?
>   260 |                         kem_alg = OQS_KEM_alg_saber_saber;
>       |                                   ^~~~~~~~~~~~~~~~~~~~~~~
>       |                                   OQS_KEM_alg_kyber_768
> oqs_kem.c:263:35: error: ‘OQS_KEM_alg_saber_firesaber’ undeclared (first use in this function)
>   263 |                         kem_alg = OQS_KEM_alg_saber_firesaber;
>       |                                   ^~~~~~~~~~~~~~~~~~~~~~~~~~~
> make[5]: *** [Makefile:618: oqs_kem.lo] Error 1
> make[5]: Leaving directory '/usr/src/strongswan-6.0beta2/src/libstrongswan/plugins/oqs'
> make[4]: *** [Makefile:2260: install-recursive] Error 1
> make[4]: Leaving directory '/usr/src/strongswan-6.0beta2/src/libstrongswan'
> make[3]: *** [Makefile:2419: install] Error 2
> make[3]: Leaving directory '/usr/src/strongswan-6.0beta2/src/libstrongswan'
> make[2]: *** [Makefile:540: install-recursive] Error 1
> make[2]: Leaving directory '/usr/src/strongswan-6.0beta2/src'
> make[1]: *** [Makefile:605: install-recursive] Error 1
> make[1]: Leaving directory '/usr/src/strongswan-6.0beta2'
> make: *** [Makefile:913: install] Error 2

Also exported the _LD_LIBRARY_PATH_ variable to _/home/umair/liboqs/build/lib_ and did `ldconfig`, not sure if it is necessary or not!

**Comment by tobiasbrunner** [comment, marked as answer]
What version of `liboqs` did you use? The latest release, 0.7.2, should definitely work.

**Reply by strongX509** [reply]
The liboqs project is currently eliminating NIST round 3 KEM algorithms that didn't make it to final or round 4 status from its master repository. As Tobias mentioned above, for the time being use version 0.7.2 of the liboqs library. As soon as liboqs 0.8 is released, we are going to adapt our selection of KEM algorithms in the strongSwan oqs plugin.

**Reply by tobiasbrunner** [reply]
Could we add `#ifdef`s there and in `get_features()` to determine the supported algorithms (similar to other crypto plugins)?

**Reply by strongX509** [reply]
I think we should wait until the liboqs 0.8 release because I expect quite some algorithm name and parameter changes by NIST and also a reduction of the algorithm variants. This would then be the time to release strongSwan 6.0.0 using KE identifiers taken from the private range.

**Reply by abc1233user** [reply]
Thank you for all the responses and suggestions.
Tried it with 0.7.2 version of liboqs and got it built successfully!

**Reply by gr4yg00s3** [reply]
Just noting here that liboqs 0.8 has been released https://github.com/open-quantum-safe/liboqs/releases/tag/0.8.0
